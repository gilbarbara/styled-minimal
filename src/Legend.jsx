import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { getTheme, isDefined, px } from './utils/helpers';
import { baseStyles } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { color: cl, fontFamily, fontWeight: fw, lineHeight, mb } = props;
  const { color, fontWeight, marginBottom } = getTheme(props, 'legend');

  return css`
    color: ${cl || color};
    display: block;
    font-family: ${fontFamily || 'inherit'};
    font-weight: ${fw || fontWeight};
    line-height: ${lineHeight || baseStyles.lineHeight};
    margin-bottom: ${px(isDefined(mb) ? mb : marginBottom)};
    white-space: nowrap;
  `;
};

const Legend = styled(Box)`
  ${styles};
`;

Legend.displayName = 'Legend';

Legend.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  ...Box.basePropTypes,
};

Legend.defaultProps = {
  as: 'legend',
};

export default Legend;
