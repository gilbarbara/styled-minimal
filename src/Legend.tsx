import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, lineHeightStyles, sx, system } from './utils/system';

import { LegendProps } from './types';

export const StyledLegend = styled('legend')<LegendProps>(
  props => {
    const { color, marginBottom } = getTheme(props, 'legend');

    return css`
      ${baseStyles};
      color: ${color};
      display: block;
      font-family: inherit;
      line-height: ${lineHeightStyles};
      margin-bottom: ${px(marginBottom)};
      white-space: nowrap;
    `;
  },
  sx,
  system,
);

export const Legend = React.forwardRef<HTMLLegendElement, LegendProps>((props, ref) => (
  <StyledLegend ref={ref} {...props} />
));
