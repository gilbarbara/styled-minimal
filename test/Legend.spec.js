import React from 'react';
import { mount } from 'enzyme';

import Legend from '../src/Legend';

describe('Legend', () => {
  const wrapper = mount(<Legend>Far far away</Legend>);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
