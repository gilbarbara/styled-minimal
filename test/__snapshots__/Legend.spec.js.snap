// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  color: #343434;
  display: block;
  font-family: inherit;
  font-weight: bold;
  line-height: 1.4;
  margin-bottom: 8px;
  white-space: nowrap;
}

<Legend
  as="legend"
>
  <legend
    className="c0"
  >
    Far far away
  </legend>
</Legend>
`;
